redash/models/__init__.py

```diff
@@ -966,18 +966,27 @@ def evaluate(self):
             else:
                 selector = self.options["selector"]
 
-            if selector == "max":
-                max_val = float("-inf")
-                for i in range(0, len(data["rows"])):
-                    max_val = max(max_val, data["rows"][i][self.options["column"]])
-                value = max_val
-            elif selector == "min":
-                min_val = float("inf")
-                for i in range(0, len(data["rows"])):
-                    min_val = min(min_val, data["rows"][i][self.options["column"]])
-                value = min_val
-            else:
-                value = data["rows"][0][self.options["column"]]
+            try:
+                if selector == "max":
+                    max_val = float("-inf")
+                    for i in range(len(data["rows"])):
+                        max_val = max(max_val, data["rows"][i][self.options["column"]])
+                    value = max_val
+                elif selector == "min":
+                    min_val = float("inf")
+                    for i in range(len(data["rows"])):
+                        min_val = min(min_val, data["rows"][i][self.options["column"]])
+                    value = min_val
+                elif selector == "avg":
+                    total_sum = 0
+                    for i in range(len(data["rows"])):
+                        total_sum += data["rows"][i][self.options["column"]]
+                    value = total_sum / len(data["rows"])
+                else:
+                    value = data["rows"][0][self.options["column"]]
+
+            except Exception:
```

Comment:
Do we know what kinds of exceptions might be raised? If so we should catch them (`TypeError` for example)

Reply:
I updated the code a little. Now, when there's an exception, it will be a ValueError. I changed line 988 accordingly